### ISSUE TITLE
.new_StudySpecification() problem with certain long formulas

### ISSUE THREAD

**Issue by benthestatistician** [benbhansen-stats/propertee#203]
Certain longer formulas passed to `*_spec()` get broken into multiple lines, causing problems for  these and [adjacent lines of `.new_StudySpecification()`](https://github.com/benbhansen-stats/propertee/blob/2a35f302b76db3ffa2d4c80c51a2aa20ead33ad7/R/StudySpecification.R#L99-L108):
```r
  if (grepl("unit_of_assignment\\([a-zA-Z]", deparse(form)) |
        grepl("uoa\\([a-zA-Z]", deparse(form))) {
```

Examples:
```r
data(STARdata)
STARdata$star_kinderg <- STARdata$stark
#' these all work fine:
rct_spec(star_kinderg ~ unit_of_assignment(studentid)+
             block(schoolidk),
         data=STARdata, na.fail=FALSE)
rct_spec(stark ~ unit_of_assignment(studentid)+
             block(schoolidk, gender),
         data=STARdata, na.fail=FALSE)
rct_spec(star_kinderg ~ uoa(studentid)+
             block(schoolidk, gender),
         data=STARdata, na.fail=FALSE)

#' this doesn't:
rct_spec(star_kinderg ~ unit_of_assignment(studentid)+
             block(schoolidk, gender),
         data=STARdata, na.fail=FALSE)

Error in if (grepl("unit_of_assignment\\([a-zA-Z]", deparse(form)) | grepl("uoa\\([a-zA-Z]",  : 
  the condition has length > 1
```
Comparing just the formulas, the first three are each one-liners but the fourth spans two. 
```r
> deparse(star_kinderg ~ unit_of_assignment(studentid)+
+ block(schoolidk))
[1] "star_kinderg ~ unit_of_assignment(studentid) + block(schoolidk)"
> deparse(stark ~ unit_of_assignment(studentid)+
+ block(schoolidk, gender))
[1] "stark ~ unit_of_assignment(studentid) + block(schoolidk, gender)"
> deparse(star_kinderg ~ uoa(studentid)+
+ block(schoolidk, gender))
[1] "star_kinderg ~ uoa(studentid) + block(schoolidk, gender)"
> deparse(star_kinderg ~ unit_of_assignment(studentid)+
+ block(schoolidk, gender))
[1] "star_kinderg ~ unit_of_assignment(studentid) + block(schoolidk, "
[2] "    gender)"        
```

**Comment by benthestatistician** [comment]
Increasing the `width_cutoff=` parameter to `deparse()` would circumvent this particular issue.
```r
deparse(star_kinderg ~ unit_of_assignment(studentid)+
+ block(schoolidk, gender), width.cutoff=100L)
[1] "star_kinderg ~ unit_of_assignment(studentid) + block(schoolidk, gender)"
```
But this seems a bit brittle.  Looking at `help("deparse")`, R version 4.0 introduced a `deparse1()` to address problems like this.  How about bumping our minimum R version up to 4.0.0, from 2.10, so we can use that?  @jwasserman2 @josherrickson

**Comment by josherrickson** [comment]
Funny enough, I ran across the exact same issue in another project last week.

1. I fully support bumping the minimum version; I might even suggest 4.2.0 which supports `|>` in case anyone wants to use that over `%>%`. (4.1.0 technically introduced the pipe, but 4.2.0 added a lot of features for it.)
2. If you don't want to increase the minimum version, all `deparse1` is doing is passing the output of `deparse` to `paste(..., collapse = " ")`, so we could implement it manually.
3. Before we just update one deparse with another, it's likely that a more structured solution with `terms` would be more future-proof. I'll take a look at it this week.

**Comment by benthestatistician** [comment]
Good point about pipes, which may be useful in vignettes.  I don't think this in itself demands much more of our attention, if there are objections to bumping the min version number to 4.0 or higher.  But perhaps you see other benefits to a "more structured solution," in which case I'm all for it.

**Comment by josherrickson** [comment]
Turns out we're using `deparse` in far more places than I anticipated. For the sake of efficiency, I think I'll just s/deparse/deparse1/ and bump the minimum version to 4.0.0.

1. Any objections?
2. Any objection to minimum version being 4.2.0?

**Comment by benthestatistician** [comment]
1\. No objections here to the version bump (although given my last message that question may have been directed to @jwasserman2).  
2\.I'd somewhat prefer to start with the bump to 4.0.0, along with a commit note pointing to this issue.  (Even if we wind up bumping further to 4.2.0 in a couple of days, perhaps in the context of a commit adding the use of the `|>` pipe to some example or another.)

**Comment by josherrickson** [comment]
Sounds good. I'll move forward; can always roll back changes and switch to the `paste` approach if @jwasserman2 or others brings objections.
